crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -536,13 +536,22 @@ fn check_type<T: TypeChecker>(binding: &Binding, semantic: &SemanticModel) -> bo
         },
 
         BindingKind::Argument => match binding.statement(semantic) {
-            // ```python
-            // def foo(x: annotation):
-            //   ...
-            // ```
-            //
-            // We trust the annotation and see if the type checker matches the annotation.
             Some(Stmt::FunctionDef(ast::StmtFunctionDef { parameters, .. })) => {
+                // ```python
+                // def foo(**kwargs):
+                //   ...
+                // ```
+                if let Some(kwarg_parameter) = parameters.kwarg.as_deref() {
+                    if kwarg_parameter.name.range() == binding.range() {
+                        return true;
```

Comment:
I don't know anything about the type checking code so I might be wrong here but doesn't this now return true for all type checkers? 

Looking at the definition of `is_dict` it returns whatever `check_types` returns and so do all other `is_*` methods:

https://github.com/astral-sh/ruff/blob/aa0db338d935aa40fafaa05feff7343c42491d8e/crates/ruff_python_semantic/src/analyze/typing.rs#L743-L745

But we could move this check into `is_dict` (and add support for `*args` to `is_list`?). But I don't know if this is idiomatic way to do this.

Reply:
Oh I see what you're saying... eek..

Reply:
I moved it into `is_dict` — I didn't want to touch the trait for this. We perform an extra `statement` lookup now but I don't think that's expensive.

Reply:
And opened a corresponding `is_tuple` fix https://github.com/astral-sh/ruff/pull/13512